### PR TITLE
Update add reading error handling

### PR DESCRIPTION
**Pull request by wumbabum**
Update api controller to be able to respond correctly to requests with invalid params with a 400.

### CODE REVIEW

File: lib/ambi.ex [hunk 1]
```diff
@@ -14,12 +14,13 @@ defmodule Ambi do
   alias Phoenix.PubSub
 
   def add_reading(attrs \\ %{}) do
-    %Ambi.Reading{}
+    result = %Ambi.Reading{}
```
Comment (Serneum) [inline lib/ambi.ex:17]: I forget, does Elixir return the last line by default? If so, you could avoid thid assign and just move the logic below the log line

File: lib/ambi_web/views/api_view.ex [hunk 2]
```diff
@@ -3,6 +3,7 @@ defmodule AmbiWeb.ApiView do
 
   # Renders a JSON response to the IoT device doing a POST
   def render("api_response.json", %{status: status}) do
+    IO.puts("render/2 ApiView")
```
Comment (Serneum) [inline lib/ambi_web/views/api_view.ex:6]: Is this purely for debugging? Can it be removed?

**Review by Serneum — COMMENTED** [review]
I'd clean up the commits a bit, but otherwise this looks good.

File: lib/ambi.ex [hunk 3]
```diff
@@ -14,12 +14,13 @@ defmodule Ambi do
...
     |> Ambi.Reading.changeset(attrs)
     |> Repo.insert()
     #broadcast_change(:added)
     #Logger.debug "Added new reading and called broadcast_change()"
```
Comment (jhodapp) [inline lib/ambi.ex:21]:
```suggestion
```

File: lib/ambi_web/views/api_view.ex (hunk 2 above)
Comment (jhodapp) [inline lib/ambi_web/views/api_view.ex:6]:
```suggestion
```

**Review by jhodapp — APPROVED** [review]
LGTM

**Comment by jhodapp** [comment]
Merging as is to clear out this old but still useful addition.

**Pull request merged**